issues with 404 without index.php

danrcoull opened the pull request:
<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description
unset index, on route check return false if not required. 
check the request uri to make sure nothing is passed or if admin is passed


### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

danrcoull committed:
issues with 404 without index.php

unset index, on route check return false if not required. 
check the request uri to make sure nothing is passed or if admin is passed

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=18378) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=18378) before we can accept your contribution.<br/>

sidolov commented on phpserver/router.php:
```diff
@@ -39,13 +39,23 @@
 
 if (php_sapi_name() === 'cli-server') {
     $debug("URI: {$_SERVER["REQUEST_URI"]}");
+    if($_SERVER["REQUEST_URI"] == '/'|| strpos($_SERVER["REQUEST_URI"],'/admin') !==false){
```
We can't hardcode `/admin` path because it can be different according to store settings

danrcoull replied:
Though we could pull directly from the config like below. 
```
$admin = isset($etcConfig['backend']['frontName']) ? $etcConfig['backend']['frontName'] : 'NO-CONFIG';

 if($_SERVER["REQUEST_URI"] == '/'|| strpos($_SERVER["REQUEST_URI"],$admin) !==false)
 {
    return false;
}
```

sidolov commented:
Hi @danrcoull  , please, sign CLA, otherwise, we can't process your pull request

danrcoull committed:
updates requested

danrcoull committed:
Wrong Update

Update to check for env config, check for backend and set the admin name to ignore any route configs.

danrcoull commented:
Ive made some modds to the code, to check the config and check if it exists to get the backend name.

Commit:
Featured: Add cms category blocks to0 graphql query
